core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -336,17 +336,27 @@ read_leb(const uint8 *buf, uint32 *p_offset, uint32 maxbits, bool sign)
         frame_sp += 2;                    \
     } while (0)
 
-#define PUSH_CSP(_label_type, param_cell_num, cell_num, _target_addr) \
-    do {                                                              \
-        bh_assert(frame_csp < frame->csp_boundary);                   \
-        /* frame_csp->label_type = _label_type; */                    \
-        frame_csp->cell_num = cell_num;                               \
-        frame_csp->begin_addr = frame_ip;                             \
-        frame_csp->target_addr = _target_addr;                        \
-        frame_csp->frame_sp = frame_sp - param_cell_num;              \
-        frame_csp++;                                                  \
+/* in exception handling, label_type needs to be stored to lookup exception
+ * handlers */
+
+#if WASM_ENABLE_EXCE_HANDLING != 0
+#define SET_LABEL_TYPE(_label_type) frame_csp->label_type = _label_type
+#else
+#define SET_LABEL_TYPE(_label_type) (void)0
+#endif
+
+#define PUSH_CSP(_label_type, param_cell_num, cell_num, _target_addr)   \
+    do {                                                                \
+        bh_assert(frame_csp < frame->csp_boundary);                     \
+        SET_LABEL_TYPE(_label_type);                                    \
+        frame_csp->cell_num = cell_num;                                 \
+        frame_csp->begin_addr = frame_ip;                               \
+        frame_csp->target_addr = _target_addr;                          \
+        frame_csp->frame_sp = frame_sp - param_cell_num;                \
+        frame_csp++;                                                    \
     } while (0)
 
+
```

Comment:
Had better not add the line if not needed.

Reply:
removed line